pkg/oc/cli/observe/observe.go

```diff
@@ -817,136 +854,54 @@ func (lw restListWatcher) Watch(opt metav1.ListOptions) (watch.Interface, error)
-		strict:           strict,
-		rawTemplates:     templates,
-		buf:              &bytes.Buffer{},
-	}
-	for _, s := range templates {
-		t := template.New("template")
-		child, err := t.Parse(s)
-		if err != nil {
-			return nil, err
-		}
-		if !strict {
-			child.Option("missingkey=zero")
-		}
-		p.templates = append(p.templates, child)
-	}
-	return p, nil
-}
-
-func (p *GoTemplateColumnPrinter) Print(obj interface{}) ([]string, []byte, error) {
-	var columns []string
-	for i, t := range p.templates {
-		p.buf.Reset()
-		if err := t.Execute(p.buf, obj); err != nil {
-			return nil, nil, fmt.Errorf("error executing template '%v': '%v'\n----data----\n%+v\n", p.rawTemplates[i], err, obj)
-		}
-		// if the template resolves to the special <no value> result, return it as an empty string
-		// most arguments will prefer empty vs an arbitrary constant, and we are making gotemplates consistent with
-		// jsonpath
-		if p.buf.String() == "<no value>" {
-			if p.strict {
-				return nil, nil, fmt.Errorf("error executing template '%v': <no value>", p.rawTemplates[i])
-			}
-			columns = append(columns, "")
-		} else {
-			columns = append(columns, p.buf.String())
-		}
-	}
-	return columns, nil, nil
-}
-
-type ColumnPrinter interface {
-	Print(obj interface{}) ([]string, []byte, error)
-}
-
 // VersionedPrinter takes runtime objects and ensures they are converted to a given API version
-// prior to being passed to a nested printer.
+// prior to being passed to a nested set of printers.
 type VersionedColumnPrinter struct {
-	printer   ColumnPrinter
-	convertor runtime.ObjectConvertor
-	version   runtime.GroupVersioner
+	printerList []printers.ResourcePrinter
+	convertor   runtime.ObjectConvertor
+	version     runtime.GroupVersioner
 }
 
 // NewVersionedHumanReadablePrinter wraps a printer to convert objects to a known API version prior to printing.
-func NewVersionedColumnPrinter(printer ColumnPrinter, convertor runtime.ObjectConvertor, version runtime.GroupVersioner) ColumnPrinter {
+func NewVersionedColumnPrinter(printerList []printers.ResourcePrinter, convertor runtime.ObjectConvertor, version runtime.GroupVersioner) *VersionedColumnPrinter {
 	return &VersionedColumnPrinter{
-		printer:   printer,
-		convertor: convertor,
-		version:   version,
+		printerList: printerList,
+		convertor:   convertor,
+		version:     version,
 	}
 }
 
-// Print converts the object to a map[string]interface{} in the target version before calling the nested printer.
-func (p *VersionedColumnPrinter) Print(out interface{}) ([]string, []byte, error) {
+func (p *VersionedColumnPrinter) PrintObj(obj runtime.Object) ([]string, []byte, error) {
 	var output []byte
-	if obj, ok := out.(runtime.Object); ok {
-		converted, err := p.convertor.ConvertToVersion(obj, p.version)
-		if err != nil {
-			if !runtime.IsNotRegisteredError(err) {
-				return nil, nil, err
-			}
-			converted = obj
-		}
-		data, err := json.Marshal(converted)
-		if err != nil {
```

Comment:
Why you needed to change this conversion logic? Is it not working? If not, drop `convertor` and `version` fields from `VersionedColumnPrinter` struct.

Reply:
We are still getting internal versioned objects from the resource.RESTClient that this command uses. Once we switch to using a dynamic client, for example, this conversion should no longer be needed as the object that the printer receives can be assumed to be external then

Reply:
This will be resolved in #20992.